tracer/src/Datadog.Trace/DuckTyping/DuckType.Statics.cs

```diff
@@ -185,6 +187,74 @@ static ModuleBuilder CreateModuleBuilder(string name, Assembly targetAssembly)
             }
         }
 
+        private static Type? GetTypeFromPartialName(string partialName, bool throwOnError = false)
```

Comment:
nit: could we split this into two? something a bit like this? So we can ignore the horrible hackiness in general (plus makes it inlining friendly etc).

Also, we don't _think_ we're going to hit the expensive edge-case, but we don't actually _know_ how often we're falling through this safely and silently... 😬 

```csharp
        private static Type? GetTypeFromPartialName(string partialName, bool throwOnError = false)
        {
            var type = Type.GetType(partialName, throwOnError: false);
            return type ?? GetTypeTheExpensiveWay(partialName, throwOnError);
        }
```

Reply:
done in [5bcd7ae](https://github.com/DataDog/dd-trace-dotnet/pull/6284/commits/5bcd7aef97145bdad1950e684d0738196e282ef4)